File: models/bloodbank/BloodIssue.js
// models/BloodIssue.js
import mongoose from "mongoose";

const BloodIssueSchema = new mongoose.Schema(
  {
    hospitalId: { type: mongoose.Schema.Types.ObjectId, ref: "Hospital", required: true },
    branchId: { type: mongoose.Schema.Types.ObjectId, ref: "Branch", required: true },

    requestId: { type: mongoose.Schema.Types.ObjectId, ref: "BloodRequest", required: true },
    componentId: { type: mongoose.Schema.Types.ObjectId, ref: "BloodComponent", required: true },

    issuedBy: { type: mongoose.Schema.Types.ObjectId, ref: "User" },
    issueDate: Date,

    status: {
      type: String,
      enum: ["Issued", "Returned"],
      default: "Issued",
    },
  },
  { timestamps: true }
);

export default mongoose.model("BloodIssue", BloodIssueSchema);

File: jstojv/src/main/java/com/samrat/jstojv/entity/BloodIssue.java
package com.samrat.jstojv.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "blood_issues")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class BloodIssue extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "hospital_id", nullable = false)
    private Hospital hospital;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "branch_id", nullable = false)
    private Branch branch;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "request_id", nullable = false)
    private BloodRequest bloodRequest;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "inventory_id")
    private BloodInventory bloodInventory;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "component_id", nullable = false)
    private BloodComponent component;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "issued_by")
    private User issuedBy;

    @Column(name = "issue_date")
    private LocalDateTime issueDate;

    @Enumerated(EnumType.STRING)
    private Status status = Status.Issued;

    public enum Status {
        Issued, Returned
    }
}
